zipline/pipeline/factors/technical.py

```diff
@@ -101,6 +102,34 @@ def compute(self, today, assets, out, data):
         out[:] = nanmean(data, axis=0)
 
 
+class LinearWeightedMovingAverage(CustomFactor, SingleInputMixin):
+    """
+    Weighted Average Value of an arbitrary column
+
+    **Default Inputs**: None
+
+    **Default Window Length**: None
+    """
+    # numpy's nan functions throw warnings when passed an array containing only
+    # nans, but they still returns the desired value (nan), so we ignore the
+    # warning.
+    ctx = ignore_nanwarnings()
+
+    def compute(self, today, assets, out, data):
+        num_days = data.shape[0]
+
+        # Initialize weights array
+        weights = np.arange(1, num_days + 1, dtype=float)
+
+        # Compute normalizer
+        normalizer = (num_days * (num_days + 1)) / 2
+
+        for i in range(data.shape[1]):
```

Comment:
We can take advantage of numpy broadcasting rules here and vectorize this loop. It looks like we want to multiply each column by the weights array. We can get numpy to do the looping in C by turning our weights into a column vector and then multiplying it by the array like:
```python
data * weights[:, np.newaxis]
```

We can also get `nansum` to loop over these columns in C by passing the `axis` keyword argument to sum over a column. For example:
```python
nansum(data * weights[:, np.newaxis], axis=0)
```

This will return a 1d array of length `data.shape[1]` which is the weighted sum of the columns. Finally, we can do the division by the normalizer in constant space by doing an inplace div:
```python
out[:] = nansum(data * weights[:, np.newaxis], axis=0)
out /= normalizer
```

This does the same as the loop but should run much faster.